src/test/java/de/wejago/hichi2influxDB/unit/MqttSubscriberServiceTest.java

```diff
@@ -0,0 +1,167 @@
+import com.influxdb.client.write.Point;
+import de.wejago.hichi2influxDB.dto.SensorEntry;
+import de.wejago.hichi2influxDB.dto.SensorMeasurement;
+import de.wejago.hichi2influxDB.service.InfluxDbService;
+import de.wejago.hichi2influxDB.service.MqttSubscriberService;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import org.eclipse.paho.client.mqttv3.IMqttClient;
+import org.eclipse.paho.client.mqttv3.IMqttMessageListener;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.internal.util.reflection.Whitebox;
+import org.springframework.boot.test.system.CapturedOutput;
+import org.springframework.boot.test.system.OutputCaptureExtension;
+
+@ExtendWith(OutputCaptureExtension.class)
+public class MqttSubscriberServiceTest {
+    @Mock
+    private InfluxDbService influxDbService;
+
+    @Mock
+    private IMqttClient mqttClient;
+    @InjectMocks
+    private MqttSubscriberService mqttSubscriberService;
+
+    private SensorEntry testSensorEntry;
+
+    private MqttMessage mqttMessage;
+
+    private Point point;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        point = generateTestPoint();
+    }
+
+    @Test
+    void testConnectSuccessfully(CapturedOutput output) throws MqttException {
+        // GIVEN
+        Whitebox.setInternalState(mqttSubscriberService, "brokerIp", "localhost");
+        Whitebox.setInternalState(mqttSubscriberService, "brokerPort", "1883");
+        Whitebox.setInternalState(mqttSubscriberService, "username", "user123");
+        Whitebox.setInternalState(mqttSubscriberService, "password", "test-pass");
+
+        // WHEN
+        mqttSubscriberService.connect();
+
+        // THEN
+        assertThat(output.getOut()).contains("Successfully connected to MQTT client!");
+    }
+
+    @Test
+    void testConnectFailureOnWrongPort(CapturedOutput output) {
+        // GIVEN
+        MqttSubscriberService mqttSubscriberService = new MqttSubscriberService();
+        Whitebox.setInternalState(mqttSubscriberService, "brokerIp", "localhost");
+        Whitebox.setInternalState(mqttSubscriberService, "brokerPort", "8080");
+        Whitebox.setInternalState(mqttSubscriberService, "username", "user123");
+        Whitebox.setInternalState(mqttSubscriberService, "password", "test-pass");
+
+        // WHEN
+        assertDoesNotThrow(() -> mqttSubscriberService.connect(), "The connect() method should not throw any exceptions");
+
+        // THEN
+        assertThat(output.getOut()).contains("Error connecting to MQTT client!");
```

Comment:
We need some error handling and retry for the MQTT connection and probably also Influx.
I would say leave it for now and we will solve that in a later task, when it is running in docker.
Both problems should make the health state change after some retries for example.

Reply:
will leave it for next task